Neos.ContentRepository/Classes/Domain/Factory/NodeTypeConstraintFactory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Domain\Factory;
```

Comment:
General question: Do we still group classes by pattern (Entity, Factory, ValueObject) or by (sub)context? I've taken the latter approach for the DimensionSpace package and am quite happy with it...

Reply:
I don't have a specific opinion on this one :)

Reply:
I'd go with context > technical pattern, but the question is: what defines a (sub)context?
@nezaniel Where would you suggest to move this one to for example?